mountpoint-s3-crt/src/s3/client.rs

```diff
@@ -539,9 +540,9 @@ impl MetaRequest {
         }
     }
 
-    /// Write a chunk of data and indicate whether it is the last. If invoked before the previous
-    /// write completed, or after setting `eof` to `true`, will return an AWS_ERROR_INVALID_STATE error.
-    pub fn write<'a>(&'a mut self, slice: &'a [u8], eof: bool) -> MetaRequestWrite<'a> {
+    /// Write a chunk of data and indicate whether it is the last. It will return an
+    /// AWS_ERROR_INVALID_STATE error if invoked after setting `eof` to `true`.
+    pub fn write<'r, 's: 'r>(&'r mut self, slice: &'s [u8], eof: bool) -> MetaRequestWrite<'r, 's> {
```

Comment:
I've simplified the lifetime parameters: we want `MetaRequestWrite` to be bound by both the request and the slice, but the lifetimes of request and slice are not otherwise related.

Reply:
I see, it's just `'s: 'r` syntax was new to me, but thanks for simplifying anyway